### DISCUSSION TITLE
How to refetch apis which is fetching?

### DISCUSSION THREAD

**Discussion by fardad-dev** [TanStack/query#1715] [Q&A]
How can I refech apis which is fetching

**Comment by TkDodo** [comment]
I don’t understand the question, sorry. Do you have an example maybe?

**Reply by fardad-dev** [reply]
When a request is fetching, I want to re-fetch that. Mean cancel in progress request and recall that again.

**Reply by TkDodo** [reply]
useQuery returns a `refetch` function, I guess you can just call that. For query cancellation, see: https://react-query.tanstack.com/guides/query-cancellation

**Reply by fardad-dev** [reply]
The refetch not working when request is in progress.

**Reply by TkDodo** [reply]
I can take a look if you provide a minimal codesandbox example that shows the problem.
